spyder/widgets/variableexplorer/dataframeeditor.py

```diff
@@ -676,10 +831,85 @@ def setup_and_check(self, data, title=''):
+
+        # Create the view for the horizontal header
+        self.table_header = QTableView()
+        self.table_header.verticalHeader().hide()
+        self.table_header.setEditTriggers(QTableWidget.NoEditTriggers)
+        self.table_header.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.table_header.setVerticalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.table_header.setHorizontalScrollMode(QTableView.ScrollPerPixel)
+        self.table_header.setHorizontalScrollBar(self.hscroll)
+        self.table_header.setFrameStyle(QFrame.Plain)
+        self.table_header.horizontalHeader().sectionResized.connect(
+                                                        self._column_resized)
+        self.table_header.setItemDelegate(QItemDelegate())
+        self.layout.addWidget(self.table_header, 0, 1)
+
+        # Create the view for the vertical index
+        self.table_index = QTableView()
+        self.table_index.horizontalHeader().hide()
+        self.table_index.setEditTriggers(QTableWidget.NoEditTriggers)
+        self.table_index.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.table_index.setVerticalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.table_index.setVerticalScrollMode(QTableView.ScrollPerPixel)
+        self.table_index.setVerticalScrollBar(self.vscroll)
+        self.table_index.setFrameStyle(QFrame.Plain)
+        self.table_index.verticalHeader().sectionResized.connect(
+                                                            self._row_resized)
+        self.table_index.setItemDelegate(QItemDelegate())
+        self.layout.addWidget(self.table_index, 1, 0)
+        self.table_index.setContentsMargins(0, 0, 0, 0)
+
+        # Create the model and view of the data
         self.dataModel = DataFrameModel(data, parent=self)
-        self.dataTable = DataFrameView(self, self.dataModel)
+        self.dataTable = DataFrameView(self, self.dataModel,
+                                       self.table_header.horizontalHeader(),
+                                       self.hscroll, self.vscroll)
+        self.dataTable.verticalHeader().hide()
+        self.dataTable.horizontalHeader().hide()
+        self.dataTable.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.dataTable.setVerticalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+        self.dataTable.setHorizontalScrollMode(QTableView.ScrollPerPixel)
+        self.dataTable.setVerticalScrollMode(QTableView.ScrollPerPixel)
+        self.dataTable.setFrameStyle(QFrame.Plain)
+        self.dataTable.setItemDelegate(QItemDelegate())
+        self.layout.addWidget(self.dataTable, 1, 1)
+        self.setFocusProxy(self.dataTable)
+        self.dataTable.sig_sortByColumn.connect(self._sort_update)
+        self.dataTable.sig_fetch_more_columns.connect(self._fetch_more_columns)
+        self.dataTable.sig_fetch_more_rows.connect(self._fetch_more_rows)

```

Comment:
Try to see if you can move these lines to `create_data_table` (it seems possible, but I'm not really sure).